### PR TITLE
otree: init at 0.2.0

### PR DESCRIPTION
**Pull request by 0x61nas**
## Description of changes

addition [otree](https://github.com/fioncat/otree) package, which is a command line tool to view objects (json/yaml/toml) in TUI tree widget

### try the package
```sh
nix run github:anas-contribs/nixpkgs/by-name/otree#otree
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ot/otree/package.nix
Comment (afh) [inline pkgs/by-name/ot/otree/package.nix:29]: Minor correction of grammar
```suggestion
  doCheck = false; # there are no tests
```

File: pkgs/by-name/ot/otree/package.nix
Comment (afh) [inline pkgs/by-name/ot/otree/package.nix:17]: ℹ️ I was able to successfully build on `aarch64-darwin` without this.

File: pkgs/by-name/ot/otree/package.nix
Comment (afh) [inline pkgs/by-name/ot/otree/package.nix:39]: See comment about darwin specific `buildInputs` above:
```suggestion
```

File: pkgs/by-name/ot/otree/package.nix
Comment (afh) [inline pkgs/by-name/ot/otree/package.nix:21]: This requires `darwin` being declared as an input at the top:
```suggestion

  buildInputs = lib.optional stdenv.isDarwin darwin.apple_sdks.frameworks.IOKit;
```

File: pkgs/by-name/ot/otree/package.nix
Comment (afh) [inline pkgs/by-name/ot/otree/package.nix:5]: Needed to enable building on darwin
```suggestion
, git
, darwin
```

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @0x61nas! 😃 👍 

Please find below a few comments and suggestions, mainly to add support for building on darwin.

Is there any way we could remove the dependency on git for building? Upstream seems to offer [`BUILD_OTREE_WITH_GIT_INFO`](https://github.com/fioncat/otree/blob/main/build.rs#L75) and [`CARGO_PKG_VERSION`](https://github.com/fioncat/otree/blob/main/build.rs#L81) could those be helpful here?

**Comment by 0x61nas** [comment]
> Is there any way we could remove the dependency on git for building?

in v0.1.0 i dont think so

>  Upstream seems to offer [`BUILD_OTREE_WITH_GIT_INFO`](https://github.com/fioncat/otree/blob/main/build.rs#L75) and [`CARGO_PKG_VERSION`](https://github.com/fioncat/otree/blob/main/build.rs#L81) could those be helpful here?

oh, that's would help ofc, i think with this we can remove `git` from our native build deps. but we have to wait until they release these modifications

**Comment by afh** [comment]
Maybe a TODO comment in the package regarding removal of the `git` dependency via use of [BUILD_OTREE_WITH_GIT_INFO](https://github.com/fioncat/otree/blob/main/build.rs#L75) and [CARGO_PKG_VERSION](https://github.com/fioncat/otree/blob/main/build.rs#L81) for the next release might helpful. What do you think, @0x61nas?

**Comment by 0x61nas** [comment]
good idea

File: pkgs/by-name/ot/otree/package.nix
Comment (afh) [inline pkgs/by-name/ot/otree/package.nix:23]: Maybe add a blank line between `buildInputs` and `cargoLock`? Apologies for not including in my previous suggestion.
```suggestion
  buildInputs = lib.optional stdenv.isDarwin darwin.apple_sdks.frameworks.IOKit;
  
```

Comment (0x61nas) [reply pkgs/by-name/ot/otree/package.nix:23]: fine, i'll squash all commits in one commit tho, do you wanna add your self as a maintainer?

Comment (afh) [reply pkgs/by-name/ot/otree/package.nix:23]: Thanks for the offer, @0x61nas, much appreciated, but I'll decline at this point. I might reconsider in the future…

**Review by afh — APPROVED** [review]
Great work on this, @0x61nas, and thank you for the contribution. I look forward to using otree in the future!

If you wanted to go above and beyond with this package run it file through [nixmft](http://github.com/NixOS/nixfmt)…

File: pkgs/by-name/ot/otree/package.nix
Comment (afh) [inline pkgs/by-name/ot/otree/package.nix:25]: @0x61nas My previous suggestion contained a typo (it's `apple_sdk` not `apple_sdks`), apologies for that…

```suggestion
  buildInputs = lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.IOKit;
```

**Review by afh — CHANGES_REQUESTED** [review]
See comment below…

File: pkgs/by-name/ot/otree/package.nix
Comment (0x61nas) [reply pkgs/by-name/ot/otree/package.nix:25]: oh, ok

Comment (0x61nas) [reply pkgs/by-name/ot/otree/package.nix]: NP, fixed

**Review by afh — APPROVED** [review]
Thanks for addressing this so quickly, @0x61nas, much appreciated!

**Review by afh — APPROVED** [review]
How fortunate, that version 0.2.0 was released while this PR is still under review and `leaveDotGit` and `git` in `nativeBuildInputs` are no longer necessary 🥳

If you wanted to go above and beyond with this PR, please consider formatting it using:
<code>nix run <a href='https://github.com/NixOS/nixfmt?tab=readme-ov-file#nixfmt'>github:NixOS/nixfmt</a> -- pkgs/by-name/ot/otree/package.nix</code>

**Comment by afh** [comment]
@0x61nas any idea what could cause the [ofborg-eval check failure](https://gist.github.com/GrahamcOfBorg/0f154ef661dc84a5f163aec353a38993)?

**Comment by IldenH** [comment]
Format using nixfmt-rfc-style
`nix run nixpkgs#nixfmt-rfc-style -- pkgs/by-name/ot/otree/package.nix`

File: pkgs/by-name/ot/otree/package.nix
Comment (FliegendeWurst) [inline pkgs/by-name/ot/otree/package.nix:2]: Please format with the new formatter.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by Aleksanaa** [comment]
Done in #358397

**Pull request closed without merging**
